src/main/scala/strawman/collection/Set.scala

```diff
@@ -42,9 +58,13 @@ trait SetLike[A, +C[X] <: Set[X]]
 trait SetMonoTransforms[A, +Repr]
   extends IterableMonoTransforms[A, Repr] {
 
-  def & (that: Set[A]): Repr
+  def & (that: Set[A]): Repr = this.filter(that)
+
+}
+
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
 
-  def ++ (that: Set[A]): Repr
+  def ++ (that: Set[A]): C[A]
```

Comment:
Thinking about this problem again, I ended up with the following question: do we really want to keep (immutable) `Set` invariant? Why not make it covariant like other immutable collection types? Do people often use the “intentional” facet of a `Set` very often in practice? We can anyway turn a `Set[A]` into a `[B >: A]B => Boolean`, like we do with other immutable collection types. The drawback is that we can get this `B` inferred to `Any` without noticing it… But we already have this issue with the other immutable collection types, right?